lib/wibox/container/align.lua

```diff
@@ -0,0 +1,158 @@
+---------------------------------------------------------------------------
+-- A container used to align smaller widgets into larger space.
+--
+--@DOC_wibox_container_defaults_align_EXAMPLE@
+-- @author Emmanuel Lepage Vallee &lt;elv1313@gmail.com&gt;
+-- @copyright 2016 Emmanuel Lepage Vallee
+-- @release @AWESOME_VERSION@
+-- @classmod wibox.container.align
+---------------------------------------------------------------------------
+
+local setmetatable = setmetatable
+local base = require("wibox.widget.base")
+local util = require("awful.util")
+
+local align = { mt = {} }
+
+-- Take the widget width/height and compute the position from the full
+-- width/height
+local align_fct = {
+    left   = function(_  , _   ) return 0                         end,
+    center = function(wdg, orig) return math.max(0, (orig-wdg)/2) end,
+    right  = function(wdg, orig) return math.max(0, orig-wdg    ) end,
+}
+align_fct.top, align_fct.bottom = align_fct.left, align_fct.right
+
+-- Layout this layout
+function align:layout(context, width, height)
+    if not self._private.widget or not self._private.widget._private.visible then
```

Comment:
I don't think you should check `visible` here. One reason is because it won't make a difference on the visual result. The second reason is "I don't think things will be properly redrawn if the child widget is made visible" (because the widget code doesn't know that this code tries to display the widget). The third reason is "because you are using other code's `_private`".